keepassxc-browser/background/keepass.js

```diff
@@ -1072,3 +1070,46 @@ keepass.updateDatabase = function() {
         });
     }, null);
 };
+
+// Parse the version number. Major, minor and patch are mandatory
+keepass.parseVersion = function(version) {
+    const ver = /^(\d+)\.(\d+)\.(\d+)?\.?(\d+)?-?(\w+)?/.exec(version);
```

Comment:
Couldn't you just use split()?

Reply:
That would need two splits if the version number contains `snapshot`, like `2.4.0-snapshot`.

Reply:
That should still be faster than a regex.

Reply:
I'll try to improve this a bit.